lib/pure/uri.nim

```diff
@@ -380,6 +380,57 @@ proc `$`*(u: Uri): string =
     result.add("#")
     result.add(u.anchor)
 
+type
+  Query* = distinct Table[string, seq[string]]
+  QueryValues* = distinct seq[string]
+
+proc `[]`*(query: Query, key: string): QueryValues =
+  return Table[string, seq[string]](query)[key].QueryValues
+
+proc `[]`*(query: Query, key: string, i: int): string =
+  return Table[string, seq[string]](query)[key][i]
+
+proc `[]`*(values: QueryValues, i: int): string = seq[string](values)[i]
+
+converter toString*(value: QueryValues): string =
+  ## Implicit converter which retrieves the first value in a list
+  ## of query values.
+  return seq[string](value)[0]
+
+proc parseQuery*(query: string): Query =
```

Comment:
This should be an iterator that yields the (key, value) pairs IMO. Not yet another API that encourages bloat everywhere.